src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machineList;
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                machineList = new BulldozerProducer().get();
+                break;
+
+            case "Excavator":
+                machineList = new ExcavatorProducer().get();
+                break;
+
+            case "Truck":
+                machineList = new TruckProducer().get();
+                break;
+
+            default:
+                machineList = new ArrayList<>();
```

Comment:
Don't return null or specific type List in `getAll()` when no producer has been found. It's better to return an empty List to avoid potential NullPointerExceptions or unexpected behavior.